subprojects/core/src/main/java/org/gradle/internal/jvm/UnsupportedJavaRuntimeException.java

```diff
@@ -18,15 +18,26 @@
 
 import org.gradle.api.GradleException;
 import org.gradle.api.JavaVersion;
+import org.gradle.util.DeprecationLogger;
 import org.gradle.util.GradleVersion;
 
 public class UnsupportedJavaRuntimeException extends GradleException {
+    public static final String JAVA7_DEPRECATION_WARNING = "Support for running Gradle using Java 7 has been deprecated and is scheduled to be removed in Gradle 5.0. "
+        + "Please see https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_cross_compilation for more details.";
```

Comment:
Maybe we can improve the error message a bit here, and be more explicit. Instead of "hey, RTFM", which is what the message says today, we can say:

> Support for running Gradle using Java 7 has been deprecated. It is however still possible to compile, test an run applications that use Java 7 using cross-compilation. See .... for more details.

Reply:
Well... is it too long for a console output?

Reply:
I'm not sure. I've always been thinking most error messages in Gradle are too terse. @jjohannes @bmuschko WDYT?

Reply:
I think it's OK to be more explicit here. In addition we will also want to add a link that allows a user to give feedback whether the deprecation would be a blocker for them. @eriwen Is there a way you could provide us with a URL that redirects to a specific page as mentioned by @bigdaz [here](https://github.com/gradle/gradle-private/issues/935#issuecomment-324648508)?